Dockerfile.openshift

```diff
@@ -1,9 +1,9 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.19-openshift-4.12 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.22-openshift-4.17 AS builder
 WORKDIR /external-dns-operator
 COPY . .
 RUN make build-operator
 
-FROM registry.ci.openshift.org/ocp/4.12:base
+FROM registry.ci.openshift.org/ocp/4.17:base
```

Comment:
As of now external-dns-operator repo controls the [base, builder](https://github.com/openshift/release/blob/04565dc79e8442bb415f24971bb00b2eab45deef/ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml#L9) and [build root](https://github.com/openshift/release/blob/04565dc79e8442bb415f24971bb00b2eab45deef/ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml#L6-L7) images from the repository. So when we need to bump golang or base image we submit a PR only into this repo. That means less automation from ART indeed. But we disabled it to get rid of openshift release branches created automatically. That was causing a confusion about the adherence of ExternalDNS Operator to OCP lifecycle.

As a matter of fact, `Dockerfile.openshift` [is not even used anymore](https://github.com/openshift/release/blob/04565dc79e8442bb415f24971bb00b2eab45deef/ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml#L9). I updated `Dockerfile.openshift` just to be consistent but it can be removed.

Reply:
> As a matter of fact, Dockerfile.openshift [is not even used anymore](https://github.com/openshift/release/blob/04565dc79e8442bb415f24971bb00b2eab45deef/ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml#L9). I updated Dockerfile.openshift just to be consistent but it can be removed.

Oh, so this Dockerfile isn't even used? That makes more sense why this is super outdated. Now I see the regular `Dockerfile` uses it's own non-OpenShift images, which makes my question about the ART bumps irrelevant.

Should we just delete it then or does it serve a purpose still?

Reply:
> Should we just delete it then or does it serve a purpose still?

No it doesn't. Let me remove it.